persist/fs/commitlog/iterator.go

```diff
@@ -177,3 +183,19 @@ func (i *iterator) nextReader() bool {
 	i.reader = reader
 	return true
 }
+
+// getFilesBasedOnShardTimeRanges determines which commit log files include
+// the time ranges that we're trying to bootstrap, and returns those
+func getFilesBasedOnShardTimeRanges(
+	fsopts fs.Options,
+	shardTimeRanges result.ShardTimeRanges,
+) ([]string, error) {
+	path := fs.CommitLogsDirPath(fsopts.FilePathPrefix())
+
+	if shardTimeRanges.IsEmpty() {
+		return fs.CommitLogFiles(path)
+	}
+
+	start, end := shardTimeRanges.MinMax()
```

Comment:
@prateek I'm actually starting to question if this logic is correct because it doesn't account for the write back / write forward buffer. Am I correct that I need to include the commit logs for the Min/Max timestamps, and then add one more commitlog on both ends to account for the forward/backwards writes?

Reply:
lol yeah it gets a little wonky. theres a good explanation here ~ https://github.com/m3db/m3db/blob/master/storage/cleanup.go#L184-L198